src/main/java/net/backupcup/mcde/util/IdentifierGlobTagList.java

```diff
@@ -0,0 +1,91 @@
+package net.backupcup.mcde.util;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import blue.endless.jankson.JsonArray;
+import blue.endless.jankson.JsonObject;
+import blue.endless.jankson.JsonPrimitive;
+import blue.endless.jankson.api.SyntaxError;
+import net.minecraft.util.Identifier;
+import net.minecraft.registry.Registry;
+
+public abstract class IdentifierGlobTagList<T> extends IdentifierGlobList<T> {
+    private final Map<String, List<Glob>> tags;
+
+    public abstract Registry<T> getRegistry();
+
+    protected IdentifierGlobTagList(Map<String, List<Glob>> globs, Map<String, List<Glob>> tags) {
+        super(globs);
+        this.tags = tags;
+    }
+
+
+    protected IdentifierGlobTagList(Map<Boolean, Map<String, List<Glob>>> map) {
+        this(map.get(false), map.get(true));
+    }
+
+    public IdentifierGlobTagList(String... globs) {
+        this(Arrays.stream(globs).collect(toGlobsWithTags()));
+    }
+
+    public boolean contains(Identifier id) {
+        return super.contains(id) ||
+            getRegistry().streamTags()
+            .filter(tag -> tags.getOrDefault(tag.id().getNamespace(), List.of()).stream().anyMatch(g -> g.engine().matches(tag.id().getPath())))
```

Comment:
Avoid arbitrary abbreviations. Whilst `kvp` makes sense, `g` should be more clearly defined. It appears that it is referring to the `glob, but to alleviate any potential confusion, this should be avoided.